source/distributed/tria_base.cc

```diff
@@ -405,6 +405,7 @@ namespace parallel
           is_local.add_range(cell_counter[my_subdomain],
                              cell_counter[my_subdomain + 1]);
           IndexSet is_ghost(cell_counter.back());
+          is_ghost.add_range(0, cell_counter.back());
```

Comment:
While not wrong for the parallel shared triangulation, I would like to add that the data structures of the partitioner and the vectors built on top of them are not at all optimized for this case. The ghost exchange is some form of `MPI_Allgather` (which by itself is not particularly scalable beyond 1k ranks), but doing it by point-to-point communication to *each* MPI rank is going to be crazy slow beyond 100 ranks already.

Reply:
The alternative is to introduce the notion of a halo layer of a single cell also in the case of `p:s:T` without artificial cells which I find a bit counter-intuitive. If someone wants to have a halo layer ( wants to scale beyond a couple of hundred of ranks even with `p:d:T`), he or she would activate artificial cells.

Reply:
@kronbichler How should we proceed here? In my opinion, the partioner should reflect that nature of the triangulation. In this case it means that all cells except of the locally-owned ones are ghost.

Reply:
Yes, this is not wrong with the present patch, but rather with the specific triangulation. Having to send information to more than ~100 MPI ranks hurts and would be better done with other communication routines. But I didn't want to imply we should add alternative communication routines, we should be aware that our data structures break down for the shared triangulation (just in case someone does not realize that from the beginning of duplicating the whole grid).

Reply:
These two lines are equivalent to the following, which I think would be easier to read:
```
  const IndexSet is_ghost = complete_index_set (this->n_active_cells());
```

Reply:
Maybe also use `this->n_active_cells()` in the initialization of `is_local` above, and assert that `this->n_active_cells() == cell_counter.back()`.

Reply:
As for the questions you all had above: I think it is fair to say that if there are no active cells, then *all* cells not locally owned are ghosts. That seems like the intended semantics, whether or not that is efficient. Maybe one could make a statement somewhere where the user selects for/against artificial cells that not allowing artificial cells is inefficient.

I am confused, however, why *all* cells are in the `is_ghost` partitioner. I guess the `MPI::Partitioner` removes the overlap with the `is_local` index set? If so, please rename `is_local_or_ghost`.

Reply:
I have made the changes. I will open a separate PR regarding: 

> Maybe one could make a statement somewhere where the user selects for/against artificial cells that not allowing artificial cells is inefficient.

in a moment.